rust/gateway/src/eventloop.rs

```diff
@@ -305,7 +340,7 @@ fn resolve_addresses(_: &str) -> std::io::Result<Vec<IpNetwork>> {
 }
 
 #[cfg(not(target_os = "windows"))]
-fn resolve_addresses(addr: &str) -> std::io::Result<Vec<IpNetwork>> {
+fn resolve_addresses(addr: &str) -> std::io::Result<Vec<IpAddr>> {
```

Comment:
You also need to change the signature for windows. It appears that we are not checking in CI that it at least compiles on Windows?